bundles/org.eclipse.mita.program/src/org/eclipse/mita/program/inferrer/ElementSizeInferrer.xtend

```diff
@@ -56,6 +60,29 @@ class ElementSizeInferrer {
 	public def ElementSizeInferenceResult infer(EObject obj) {
 		return obj.doInfer;
 	}
+	
+	private static class Combination extends ElementSizeInferrer {
+		private final ElementSizeInferrer i1;
+		private final ElementSizeInferrer i2;
+		new(ElementSizeInferrer i1, ElementSizeInferrer i2) {
+			super();
+			this.i1 = i1;
+			this.i2 = i2;
+		}
+		
+		public override def infer(EObject obj) {
+			i1.infer(obj).orElse([| i2.infer(obj)]);
```

Comment:
This is code that uses already implemented (*in this pull request) functionality to lift abstractions. `orElse` actually does more than your code: if both results are invalid, it concatenates their messages.

The abundant null checks are because ESI are prune to returning null due to complicated code paths. ESIs should never do that, but we check to crash less.

By defining `orElse` bottom-up, both ESIs and ESIRs can be used with it, which I find to be very handy and intuitive.